models/PyCryptoBot.py

```diff
@@ -1023,50 +1034,86 @@ def startApp(self, account, last_action='', banner=True):
                 attempts = 0
 
                 if self.simstartdate is not None and self.simenddate is not None:
-                    date = self.simstartdate.split('-')
-                    startDate = datetime(int(date[0]), int(date[1]), int(date[2]))
+												   
+                    startDate = self.getDateFromISO8601Str(self.simstartdate.replace("T", " "))
+																				   
                     if self.simenddate == 'now':
-                        endDate = datetime.now()
+                        endDate = self.getDateFromISO8601Str(str(datetime.now()).split(".")[0])
                     else:
-                        date = self.simenddate.split('-')
-                        endDate = datetime(int(date[0]), int(date[1]), int(date[2]))
-                    while len(tradingData) != 300 and attempts < 10:
-                        tradingData = self.getHistoricalData(self.getMarket(), self.getGranularity(),
-                                                             startDate.isoformat(timespec='milliseconds'),
-                                                             endDate.isoformat(timespec='milliseconds'))
+                        endDate = self.getDateFromISO8601Str(self.simenddate.replace("T", " "))
+																					 
+                    while len(tradingData) < 300 and attempts < 10:
+                        if self.smart_switch == 1:
+                            tradingData = self.getSmartSwitchHistoricalDataChained(self.market, 3600,
```

Comment:
why do we have `granularity` fixed to 1 hour? If the granularity has changed, let's change it for the whole bot, because it might be quite hard to debug situation when we get 1h data here, but the rest of the bot thinks in different granularity.

Reply:
I missed that, I will update